File: snippets/screens/compose/social_login_screen/3/social_login_screen.kt
package com.example.auth.screens

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import kotlinx.coroutines.launch

enum class SocialProvider(
    val displayName: String,
    val icon: ImageVector,
    val color: Color,
    val backgroundColor: Color = Color.White
) {
    GOOGLE("Google", Icons.Default.Language, Color(0xFFDB4437)),
    APPLE("Apple", Icons.Default.Apple, Color.Black),
    FACEBOOK("Facebook", Icons.Default.Facebook, Color(0xFF4267B2)),
    MICROSOFT("Microsoft", Icons.Default.Window, Color(0xFF0078D4)),
    TWITTER("Twitter", Icons.Default.AlternateEmail, Color(0xFF1DA1F2)),
    GITHUB("GitHub", Icons.Default.Code, Color.Black)
}

@Composable
fun SocialLoginScreen(
    viewModel: SocialLoginViewModel = viewModel(),
    onNavigateToEmailLogin: () -> Unit = {},
    onNavigateToSignup: () -> Unit = {},
    onLoginSuccess: () -> Unit = {}
) {
    val scrollState = rememberScrollState()
    val snackbarHostState = remember { SnackbarHostState() }
    val coroutineScope = rememberCoroutineScope()

    LaunchedEffect(viewModel.loginSuccess) {
        if (viewModel.loginSuccess) {
            coroutineScope.launch {
                snackbarHostState.showSnackbar("Successfully signed in!")
            }
            onLoginSuccess()
        }
    }

    Scaffold(
        snackbarHost = { SnackbarHost(snackbarHostState) }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(24.dp)
                .verticalScroll(scrollState),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(modifier = Modifier.height(60.dp))

            // App logo
            AppLogoCard()

            Spacer(modifier = Modifier.height(32.dp))

            // Header
            Text(
                text = "Welcome Back",
                style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center
            )

            Spacer(modifier = Modifier.height(8.dp))

            Text(
                text = "Choose your preferred sign-in method",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )

            Spacer(modifier = Modifier.height(40.dp))

            viewModel.error?.let { error ->
                ErrorCard(message = error)
                Spacer(modifier = Modifier.height(24.dp))
            }

            // Social login buttons
            SocialProvider.values().forEach { provider ->
                SocialLoginButton(
                    provider = provider,
                    isLoading = viewModel.isLoading && viewModel.loadingProvider == provider,
                    onClick = { viewModel.signInWith(provider) }
                )
                Spacer(modifier = Modifier.height(12.dp))
            }

            Spacer(modifier = Modifier.height(32.dp))

            // Divider
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                HorizontalDivider(modifier = Modifier.weight(1f))
                Text(
                    text = "OR",
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(horizontal = 16.dp)
                )
                HorizontalDivider(modifier = Modifier.weight(1f))
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Email login button
            Button(
                onClick = onNavigateToEmailLogin,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = MaterialTheme.colorScheme.primary
                )
            ) {
                Text(
                    text = "Sign in with Email",
                    fontWeight = FontWeight.SemiBold
                )
            }

            Spacer(modifier = Modifier.height(32.dp))

            // Privacy notice
            PrivacyNoticeCard()

            Spacer(modifier = Modifier.height(32.dp))

            // Sign up link
            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Don't have an account? ",
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                TextButton(onClick = onNavigateToSignup) {
                    Text("Sign Up")
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Terms and privacy
            TermsAndPrivacyText()

            Spacer(modifier = Modifier.height(40.dp))
        }
    }
}

@Composable
fun SocialLoginButton(
    provider: SocialProvider,
    isLoading: Boolean,
    onClick: () -> Unit
) {
    OutlinedButton(
        onClick = onClick,
        enabled = !isLoading,
        modifier = Modifier
            .fillMaxWidth()
            .height(56.dp),
        border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline),
        colors = ButtonDefaults.outlinedButtonColors(
            containerColor = provider.backgroundColor
        )
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.Start
        ) {
            if (isLoading) {
                CircularProgressIndicator(
                    modifier = Modifier.size(20.dp),
                    color = provider.color,
                    strokeWidth = 2.dp
                )
            } else {
                Icon(
                    imageVector = provider.icon,
                    contentDescription = null,
                    modifier = Modifier.size(20.dp),
                    tint = provider.color
                )
            }

            Spacer(modifier = Modifier.width(12.dp))

            Text(
                text = "Continue with ${provider.displayName}",
                style = MaterialTheme.typography.bodyLarge,
                fontWeight = FontWeight.Medium,
                color = MaterialTheme.colorScheme.onSurface,
                modifier = Modifier.weight(1f),
                textAlign = TextAlign.Start
            )
        }
    }
}

@Composable
fun AppLogoCard() {
    Card(
        modifier = Modifier.size(80.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.primary
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 8.dp)
    ) {
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = Icons.Default.AccountCircle,
                contentDescription = "App Logo",
                modifier = Modifier.size(50.dp),
                tint = MaterialTheme.colorScheme.onPrimary
            )
        }
    }
}

@Composable
fun PrivacyNoticeCard() {
    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.primaryContainer
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(
                imageVector = Icons.Default.Security,
                contentDescription = null,
                modifier = Modifier.size(24.dp),
                tint = MaterialTheme.colorScheme.onPrimaryContainer
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "Your Privacy Matters",
                style = MaterialTheme.typography.titleSmall,
                fontWeight = FontWeight.SemiBold,
                color = MaterialTheme.colorScheme.onPrimaryContainer
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = "We use secure authentication and never store your social media passwords.",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onPrimaryContainer,
                textAlign = TextAlign.Center
            )
        }
    }
}

@Composable
fun TermsAndPrivacyText() {
    Text(
        text = "By continuing, you agree to our Terms of Service and Privacy Policy",
        style = MaterialTheme.typography.bodySmall,
        color = MaterialTheme.colorScheme.onSurfaceVariant,
        textAlign = TextAlign.Center,
        modifier = Modifier.padding(horizontal = 16.dp)
    )
}

// Compact social login component for use in other screens
@Composable
fun CompactSocialLogin(
    onProviderSelected: (SocialProvider) -> Unit,
    isLoading: Boolean = false,
    modifier: Modifier = Modifier
) {
    val mainProviders = listOf(SocialProvider.GOOGLE, SocialProvider.APPLE, SocialProvider.FACEBOOK)

    Column(
        modifier = modifier
    ) {
        // Divider
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically
        ) {
            HorizontalDivider(modifier = Modifier.weight(1f))
            Text(
                text = "OR",
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = FontWeight.Medium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                modifier = Modifier.padding(horizontal = 16.dp)
            )
            HorizontalDivider(modifier = Modifier.weight(1f))
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Social buttons row
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            mainProviders.forEach { provider ->
                CompactSocialButton(
                    provider = provider,
                    isLoading = isLoading,
                    onClick = { onProviderSelected(provider) },
                    modifier = Modifier.weight(1f)
                )
            }
        }
    }
}

@Composable
fun CompactSocialButton(
    provider: SocialProvider,
    isLoading: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    OutlinedButton(
        onClick = onClick,
        enabled = !isLoading,
        modifier = modifier.height(50.dp),
        border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline),
        colors = ButtonDefaults.outlinedButtonColors(
            containerColor = provider.backgroundColor
        )
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(20.dp),
                color = provider.color,
                strokeWidth = 2.dp
            )
        } else {
            Icon(
                imageVector = provider.icon,
                contentDescription = provider.displayName,
                modifier = Modifier.size(24.dp),
                tint = provider.color
            )
        }
    }
}

// Full-width colored social buttons alternative
@Composable
fun FullWidthSocialLogin(
    providers: List<SocialProvider>,
    onProviderSelected: (SocialProvider) -> Unit,
    isLoading: Boolean = false,
    loadingProvider: SocialProvider? = null,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        providers.forEach { provider ->
            val isCurrentlyLoading = isLoading && loadingProvider == provider

            Button(
                onClick = { onProviderSelected(provider) },
                enabled = !isLoading,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = provider.color,
                    contentColor = Color.White
                )
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.Start
                ) {
                    if (isCurrentlyLoading) {
                        CircularProgressIndicator(
                            modifier = Modifier.size(20.dp),
                            color = Color.White,
                            strokeWidth = 2.dp
                        )
                    } else {
                        Icon(
                            imageVector = provider.icon,
                            contentDescription = null,
                            modifier = Modifier.size(20.dp),
                            tint = Color.White
                        )
                    }

                    Spacer(modifier = Modifier.width(12.dp))

                    Text(
                        text = "Continue with ${provider.displayName}",
                        style = MaterialTheme.typography.bodyLarge,
                        fontWeight = FontWeight.SemiBold,
                        modifier = Modifier.weight(1f),
                        textAlign = TextAlign.Start
                    )
                }
            }
        }
    }
}

class SocialLoginViewModel : androidx.lifecycle.ViewModel() {
    var isLoading by mutableStateOf(false)
        private set
    var loadingProvider by mutableStateOf<SocialProvider?>(null)
        private set
    var loginSuccess by mutableStateOf(false)
        private set
    var error by mutableStateOf<String?>(null)
        private set

    fun signInWith(provider: SocialProvider) {
        isLoading = true
        loadingProvider = provider
        error = null

        // Simulate API call
        androidx.lifecycle.viewModelScope.launch {
            try {
                kotlinx.coroutines.delay(2000)
                loginSuccess = true
            } catch (e: Exception) {
                error = "Failed to sign in with ${provider.displayName}. Please try again."
            } finally {
                isLoading = false
                loadingProvider = null
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
fun SocialLoginScreenPreview() {
    MaterialTheme {
        SocialLoginScreen()
    }
}

@Preview(showBackground = true)
@Composable
fun CompactSocialLoginPreview() {
    MaterialTheme {
        Column(modifier = Modifier.padding(24.dp)) {
            CompactSocialLogin(
                onProviderSelected = {}
            )
        }
    }
}

@Preview(showBackground = true)
@Composable
fun FullWidthSocialLoginPreview() {
    MaterialTheme {
        Column(modifier = Modifier.padding(24.dp)) {
            FullWidthSocialLogin(
                providers = listOf(SocialProvider.GOOGLE, SocialProvider.FACEBOOK, SocialProvider.APPLE),
                onProviderSelected = {}
            )
        }
    }
}

File: snippets/screens/flutter/social_login_screen/2/social_login_screen.dart
import 'package:flutter/material.dart';

enum SocialProvider { google, facebook, apple, twitter, microsoft, github }

class SocialLoginScreen extends StatefulWidget {
  const SocialLoginScreen({super.key});

  @override
  State<SocialLoginScreen> createState() => _SocialLoginScreenState();
}

class _SocialLoginScreenState extends State<SocialLoginScreen> {
  bool _isLoading = false;
  SocialProvider? _loadingProvider;
  String? _error;

  Future<void> _signInWithProvider(SocialProvider provider) async {
    setState(() {
      _isLoading = true;
      _loadingProvider = provider;
      _error = null;
    });

    try {
      // Simulate social login API call
      await Future.delayed(const Duration(seconds: 2));

      // Handle successful login
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Successfully signed in with ${_getProviderName(provider)}!')),
        );
        Navigator.of(context).pushReplacementNamed('/home');
      }
    } catch (e) {
      setState(() => _error = 'Failed to sign in with ${_getProviderName(provider)}. Please try again.');
    } finally {
      if (mounted) {
        setState(() {
          _isLoading = false;
          _loadingProvider = null;
        });
      }
    }
  }

  String _getProviderName(SocialProvider provider) {
    switch (provider) {
      case SocialProvider.google:
        return 'Google';
      case SocialProvider.facebook:
        return 'Facebook';
      case SocialProvider.apple:
        return 'Apple';
      case SocialProvider.twitter:
        return 'Twitter';
      case SocialProvider.microsoft:
        return 'Microsoft';
      case SocialProvider.github:
        return 'GitHub';
    }
  }

  IconData _getProviderIcon(SocialProvider provider) {
    switch (provider) {
      case SocialProvider.google:
        return Icons.google;
      case SocialProvider.facebook:
        return Icons.facebook;
      case SocialProvider.apple:
        return Icons.apple;
      case SocialProvider.twitter:
        return Icons.alternate_email;
      case SocialProvider.microsoft:
        return Icons.microsoft;
      case SocialProvider.github:
        return Icons.code;
    }
  }

  Color _getProviderColor(SocialProvider provider) {
    switch (provider) {
      case SocialProvider.google:
        return const Color(0xFFDB4437);
      case SocialProvider.facebook:
        return const Color(0xFF4267B2);
      case SocialProvider.apple:
        return Colors.black;
      case SocialProvider.twitter:
        return const Color(0xFF1DA1F2);
      case SocialProvider.microsoft:
        return const Color(0xFF0078D4);
      case SocialProvider.github:
        return Colors.black87;
    }
  }

  Widget _buildSocialButton({
    required SocialProvider provider,
    required VoidCallback onPressed,
  }) {
    final isCurrentlyLoading = _isLoading && _loadingProvider == provider;

    return Container(
      width: double.infinity,
      height: 56,
      margin: const EdgeInsets.only(bottom: 12),
      child: OutlinedButton.icon(
        onPressed: _isLoading ? null : onPressed,
        icon: isCurrentlyLoading
            ? const SizedBox(
                width: 20,
                height: 20,
                child: CircularProgressIndicator(strokeWidth: 2),
              )
            : Icon(
                _getProviderIcon(provider),
                color: _getProviderColor(provider),
              ),
        label: Text(
          'Continue with ${_getProviderName(provider)}',
          style: TextStyle(
            color: _isLoading ? Colors.grey : Colors.black87,
            fontWeight: FontWeight.w500,
          ),
        ),
        style: OutlinedButton.styleFrom(
          backgroundColor: Colors.white,
          side: BorderSide(color: Colors.grey[300]!),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(24.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              const SizedBox(height: 60),

              // App Logo/Icon
              Container(
                width: 80,
                height: 80,
                decoration: BoxDecoration(
                  color: Theme.of(context).primaryColor,
                  borderRadius: BorderRadius.circular(20),
                ),
                child: const Icon(
                  Icons.account_circle,
                  size: 50,
                  color: Colors.white,
                ),
              ),
              const SizedBox(height: 32),

              Text(
                'Welcome Back',
                style: Theme.of(context).textTheme.headlineMedium?.copyWith(
                  fontWeight: FontWeight.bold,
                ),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 8),

              Text(
                'Choose your preferred sign-in method',
                style: Theme.of(context).textTheme.bodyLarge?.copyWith(
                  color: Colors.grey[600],
                ),
                textAlign: TextAlign.center,
              ),

              const SizedBox(height: 40),

              if (_error != null) ...[
                Container(
                  padding: const EdgeInsets.all(16),
                  decoration: BoxDecoration(
                    color: Colors.red[50],
                    borderRadius: BorderRadius.circular(12),
                    border: Border.all(color: Colors.red[200]!),
                  ),
                  child: Row(
                    children: [
                      Icon(Icons.error_outline, color: Colors.red[700]),
                      const SizedBox(width: 12),
                      Expanded(
                        child: Text(
                          _error!,
                          style: TextStyle(color: Colors.red[700]),
                        ),
                      ),
                    ],
                  ),
                ),
                const SizedBox(height: 24),
              ],

              // Social Login Buttons
              _buildSocialButton(
                provider: SocialProvider.google,
                onPressed: () => _signInWithProvider(SocialProvider.google),
              ),

              _buildSocialButton(
                provider: SocialProvider.apple,
                onPressed: () => _signInWithProvider(SocialProvider.apple),
              ),

              _buildSocialButton(
                provider: SocialProvider.facebook,
                onPressed: () => _signInWithProvider(SocialProvider.facebook),
              ),

              _buildSocialButton(
                provider: SocialProvider.microsoft,
                onPressed: () => _signInWithProvider(SocialProvider.microsoft),
              ),

              _buildSocialButton(
                provider: SocialProvider.twitter,
                onPressed: () => _signInWithProvider(SocialProvider.twitter),
              ),

              _buildSocialButton(
                provider: SocialProvider.github,
                onPressed: () => _signInWithProvider(SocialProvider.github),
              ),

              const SizedBox(height: 32),

              // Divider
              Row(
                children: [
                  Expanded(child: Divider(color: Colors.grey[300])),
                  Padding(
                    padding: const EdgeInsets.symmetric(horizontal: 16),
                    child: Text(
                      'OR',
                      style: TextStyle(
                        color: Colors.grey[600],
                        fontWeight: FontWeight.w500,
                      ),
                    ),
                  ),
                  Expanded(child: Divider(color: Colors.grey[300])),
                ],
              ),

              const SizedBox(height: 24),

              // Email/Password Login Button
              ElevatedButton(
                onPressed: () => Navigator.of(context).pushNamed('/login'),
                style: ElevatedButton.styleFrom(
                  minimumSize: const Size(double.infinity, 56),
                  backgroundColor: Theme.of(context).primaryColor,
                  foregroundColor: Colors.white,
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(12),
                  ),
                ),
                child: const Text(
                  'Sign in with Email',
                  style: TextStyle(fontWeight: FontWeight.w600),
                ),
              ),

              const SizedBox(height: 32),

              // Privacy Notice
              Container(
                padding: const EdgeInsets.all(16),
                decoration: BoxDecoration(
                  color: Colors.blue[50],
                  borderRadius: BorderRadius.circular(12),
                  border: Border.all(color: Colors.blue[200]!),
                ),
                child: Column(
                  children: [
                    Icon(
                      Icons.security,
                      color: Colors.blue[700],
                      size: 24,
                    ),
                    const SizedBox(height: 8),
                    Text(
                      'Your Privacy Matters',
                      style: TextStyle(
                        color: Colors.blue[700],
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    const SizedBox(height: 4),
                    Text(
                      'We use secure authentication and never store your social media passwords.',
                      style: TextStyle(
                        color: Colors.blue[700],
                        fontSize: 12,
                      ),
                      textAlign: TextAlign.center,
                    ),
                  ],
                ),
              ),

              const SizedBox(height: 32),

              // Sign Up Link
              Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Text(
                    'Don\'t have an account? ',
                    style: TextStyle(color: Colors.grey[600]),
                  ),
                  TextButton(
                    onPressed: () => Navigator.of(context).pushNamed('/signup'),
                    child: const Text('Sign Up'),
                  ),
                ],
              ),

              const SizedBox(height: 16),

              // Terms and Privacy
              Text.rich(
                TextSpan(
                  style: TextStyle(
                    color: Colors.grey[600],
                    fontSize: 12,
                  ),
                  children: [
                    const TextSpan(text: 'By continuing, you agree to our '),
                    TextSpan(
                      text: 'Terms of Service',
                      style: TextStyle(
                        color: Theme.of(context).primaryColor,
                        decoration: TextDecoration.underline,
                      ),
                    ),
                    const TextSpan(text: ' and '),
                    TextSpan(
                      text: 'Privacy Policy',
                      style: TextStyle(
                        color: Theme.of(context).primaryColor,
                        decoration: TextDecoration.underline,
                      ),
                    ),
                  ],
                ),
                textAlign: TextAlign.center,
              ),

              const SizedBox(height: 20),
            ],
          ),
        ),
      ),
    );
  }
}

// Alternative compact social login widget that can be used in other screens
class CompactSocialLogin extends StatelessWidget {
  final Function(SocialProvider) onProviderSelected;
  final bool isLoading;

  const CompactSocialLogin({
    super.key,
    required this.onProviderSelected,
    this.isLoading = false,
  });

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Row(
          children: [
            Expanded(child: Divider(color: Colors.grey[300])),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16),
              child: Text(
                'OR',
                style: TextStyle(
                  color: Colors.grey[600],
                  fontWeight: FontWeight.w500,
                ),
              ),
            ),
            Expanded(child: Divider(color: Colors.grey[300])),
          ],
        ),
        const SizedBox(height: 24),

        Row(
          children: [
            Expanded(
              child: _buildCompactSocialButton(
                context: context,
                provider: SocialProvider.google,
                icon: Icons.google,
                color: const Color(0xFFDB4437),
                onPressed: () => onProviderSelected(SocialProvider.google),
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: _buildCompactSocialButton(
                context: context,
                provider: SocialProvider.apple,
                icon: Icons.apple,
                color: Colors.black,
                onPressed: () => onProviderSelected(SocialProvider.apple),
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: _buildCompactSocialButton(
                context: context,
                provider: SocialProvider.facebook,
                icon: Icons.facebook,
                color: const Color(0xFF4267B2),
                onPressed: () => onProviderSelected(SocialProvider.facebook),
              ),
            ),
          ],
        ),
      ],
    );
  }

  Widget _buildCompactSocialButton({
    required BuildContext context,
    required SocialProvider provider,
    required IconData icon,
    required Color color,
    required VoidCallback onPressed,
  }) {
    return OutlinedButton(
      onPressed: isLoading ? null : onPressed,
      style: OutlinedButton.styleFrom(
        minimumSize: const Size(0, 50),
        backgroundColor: Colors.white,
        side: BorderSide(color: Colors.grey[300]!),
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(12),
        ),
      ),
      child: Icon(
        icon,
        color: isLoading ? Colors.grey : color,
        size: 24,
      ),
    );
  }
}